apis/v1alpha1/httproute_types.go

```diff
@@ -270,6 +270,25 @@ const (
 	HeaderMatchImplementationSpecific HeaderMatchType = "ImplementationSpecific"
 )
 
+// QueryParamMatchType specifies the semantics of how HTTP query parameter
+// values should be compared. Valid QueryParamMatchType values are:
+//
+// * "Exact"
+// * "Presence"
+// * "RegularExpression"
+// * "ImplementationSpecific"
+//
+// +kubebuilder:validation:Enum=Exact;Presence;RegularExpression;ImplementationSpecific
+type QueryParamMatchType string
+
+// QueryParamMatchType constants.
+const (
+	QueryParamMatchExact                  HeaderMatchType = "Exact"
+	QueryParamMatchPresence               HeaderMatchType = "Presence"
+	QueryParamMatchRegularExpression      HeaderMatchType = "RegularExpression"
+	QueryParamMatchImplementationSpecific HeaderMatchType = "ImplementationSpecific"
```

Comment:
The types of all these should be `QueryParamMatchType`.

Reply:
🤦 thanks for catching that!